threestudio/systems/base.py

```diff
@@ -275,12 +276,16 @@ def on_fit_start(self) -> None:
             threestudio.warn(
                 f"Saving directory not set for the system, visualization results will not be saved"
             )
+        if self.cfg.geometry.shape_init:
+            self.geometry.initialize_shape()
 
     def on_test_end(self) -> None:
         if self._save_dir is not None:
             threestudio.info(f"Test results saved to {self._save_dir}")
 
     def on_predict_start(self) -> None:
+        if self.cfg.geometry.shape_init and self.cfg.geometry.fix_geometry:
+            self.geometry.initialize_shape()
```

Comment:
Why initilize_shape twice

Reply:
And what if the geometry doesn't have `shape_init` and `fix_geometry` attributes?

Reply:
You're right. I wrote this because the shape was not initialized when I export meshes. I should instead register the custom mesh to buffer.

Reply:
I don't think we need to register the mesh. The geometry is already stored in the grid which will be saved to the checkpoint. When resumed from a checkpoint, the geometry should be correctly loaded.

Reply:
@yankeesong Did you export directly without training? In that way the mesh would indeed not be initialized.

Reply:
No it is not saved to the checkpoint. If the mesh comes from a MLP then it's stored. If it's a custom mesh it is not stored. 

I tried saving to buffer but it's quite untidy and involves changing many other things. Do you guys have any other suggestions besides initializing again at validation?

Or I guess the best solution is to write a new geometry module.

Reply:
@bennyguo Yeah in general if there's any kind of pause/resume happening then the custom mesh needs to be re-initialized.